content/en/docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -321,6 +321,57 @@ Then run the following command to force decrypt all Secrets:
+
+For Kubernetes encryption at rest, this would mean that keys should be rotated at least once every two years, and a complete storage migration should be done at the minimum once every three years after the first rotation.
+
+However, depending on the encryption provider, these cryptoperiod might be too long, and other threats need to be considered.
+
+### aesgcm
+
+The `aesgcm` provider is using the AES-GCM encryption algorithm with a random nonce to encrypt data. This algorithm is well known for being vulnerable to nonce collision attacks and as such it is highly recommended to have an automated key rotation mechanism in place to use it.
+
+For this type of attack, the construction of the IV is important and the NIST recommends only supporting nonce of the length of 96 bits which is what Kubernetes does.
+
+> For IVs, it is recommended that implementations restrict support to the length of 96 bits, to promote interoperability, efficiency, and simplicity of design.
+> -- <cite>[NIST.SP.800-38D](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf#page=16)<cite>
+
+For this type of IV construction, the NIST has the following requirement for the number of invocations of the keys:
+
+> The total number of invocations of the authenticated encryption function shall not exceed $2^{32}$, including all IV lengths and all instances of the authenticated encryption function with the given key.
+> -- <cite>[NIST.SP.800-38D](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf#page=29)<cite>
+
+This key invocation requirement is calculated based on the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem). It is used to know the change of a nonce collision happening based on the number of encryptions performed by a single key.
+
+The formula provided by this problem to calculate the chance of a collision happening is as follow with `n` being the number of unique combination of value for the nonce and `k` the number of encryption done by a single-key:
+
+$$Vnr = {\frac{n!}{(n-k)!}}$$
+
+Kubernetes uses a 96 bits nonce so there are $2^{96}$ possible combinaisons
+
+$$Vnr = {\frac{2^{96}!}{(2^{96}-k)!}}$$
+
+This is a complex formula to solve with big numbers, so we will use an approximation to calculate the probability of a nonce collision happening
+
+$$p(n,k) \approx 1 - e^{-{\frac{(k^2)}{2n}}}$$
+
+$$p(k) \approx 1 - e^{-{\frac{(k^2)}{2^{97}}}}$$
+
+A negligeable chance of collision by current security standards is $1/2^{32}$ so in Kubernetes' case we want to make sure that the chance of a none collision is less than one in $2^{32}$
+
+$$p(k) \approx 1 - e^{-{\frac{(k^2)}{2^{97}}}} \leq 2^{32}$$
+
+The smallest power of 2 for which this equation is satified is $2^{32}$
+
+$$1-e^{-\frac{2^{32}}{2^{97}}} \approx 1.164\times{10}^{-10} \leq \frac{1}{2^{32}} \approx 2.32\times{10}^{-10} \leq 1-e^{-\frac{2^{33}}{2^{97}}} \approx 4.656\times{10}^{-10}$$
```

Comment:
I tried to solve the formula with `k` as the unknown and find its value based on `p(k) <= 2^32`, but couldn't get the result I wanted. For the record, the logic to solve the birthday paradox with that kind of equation is explained in https://en.wikipedia.org/wiki/Birthday_problem under `An upper bound on the probability and a lower bound on the number of people`.